components/lib/crash/src/main/java/mozilla/components/lib/crash/service/SendCrashReportService.kt

```diff
@@ -0,0 +1,81 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.lib.crash.service
+
+import android.app.Service
+import android.content.Intent
+import android.os.Build
+import android.os.IBinder
+import androidx.annotation.VisibleForTesting
+import androidx.core.app.NotificationCompat
+import androidx.core.app.NotificationManagerCompat
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.GlobalScope
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
+import mozilla.components.lib.crash.Crash
+import mozilla.components.lib.crash.CrashReporter
+import mozilla.components.lib.crash.R
+import mozilla.components.lib.crash.notification.CrashNotification
+import mozilla.components.lib.crash.notification.NOTIFICATION_TAG
+import mozilla.components.support.base.ids.cancel
+import kotlin.coroutines.CoroutineContext
+import kotlin.coroutines.EmptyCoroutineContext
+
+private const val NOTIFICATION_ID = 1
+
+class SendCrashReportService : Service() {
+    private val crashReporter: CrashReporter by lazy { CrashReporter.requireInstance }
+    private val logger by lazy { CrashReporter
+            .requireInstance
+            .logger
+    }
+
+    private var reporterCoroutineContext: CoroutineContext = EmptyCoroutineContext
+
+    override fun onStartCommand(intent: Intent, flags: Int, startId: Int): Int {
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+            val channel = CrashNotification.ensureChannelExists(this)
+            val notification = NotificationCompat.Builder(this, channel)
+                    .setContentTitle(getString(R.string.mozac_lib_crash_dialog_checkbox,
+                            crashReporter.promptConfiguration.organizationName))
+                    .setSmallIcon(R.drawable.mozac_lib_crash_notification)
+                    .setPriority(NotificationCompat.PRIORITY_DEFAULT)
+                    .setCategory(NotificationCompat.CATEGORY_ERROR)
+                    .setAutoCancel(true)
+                    .setProgress(0, 0, true)
+                    .build()
+
+            startForeground(NOTIFICATION_ID, notification)
+        }
+
+        intent.extras?.let { extras ->
+            val crash = Crash.NativeCodeCrash.fromBundle(extras)
+            NotificationManagerCompat.from(this).cancel(this, NOTIFICATION_TAG)
+
+            sendCrashReport(crash) {
+                stopSelf()
+            }
+        } ?: logger.error("Received intent with null extras")
+
+        return START_STICKY
```

Comment:
Do we really want STICKY here? My assumption is that we never want Android to re-launch this service on its own and return NOT_STICKY here? 🤔

Reply:
Done.